www/src/components/account/billing/ConfirmPayment.tsx

```diff
@@ -0,0 +1,178 @@
+import {
+  ReactNode,
+  useContext,
+  useEffect,
+  useState,
+} from 'react'
+import { Link, useSearchParams } from 'react-router-dom'
+import { Div, Flex, P } from 'honorable'
+import { Button, LoopingLogo, Modal } from '@pluralsh/design-system'
+import { useStripe } from '@stripe/react-stripe-js'
+import { SetupIntent } from '@stripe/stripe-js'
+
+import PlatformPlansContext from '../../../contexts/PlatformPlansContext'
+
+import { namedOperations, useUpgradeToProfessionalPlanMutation } from '../../../generated/graphql'
+
+import { PlanType } from './PaymentForm'
+
+export default function ConfirmPayment() {
+  const stripe = useStripe()
+  const [searchParams] = useSearchParams()
+
+  console.log('stripe', stripe)
+
+  console.log('searchParams', searchParams)
+  const clientSecret = searchParams.get('setup_intent_client_secret')
+  const plan: PlanType
+    = searchParams.get('plan') === 'yearly' ? 'yearly' : 'monthly'
+  const { proPlatformPlan, proYearlyPlatformPlan }
+    = useContext(PlatformPlansContext)
+  const planId
+    = plan === 'yearly' ? proYearlyPlatformPlan.id : proPlatformPlan.id
+
+  console.log('clientSecret', clientSecret)
+
+  const [upgradeSuccess, setUpgradeSuccess] = useState(false)
+  const [error, setError] = useState<{ message?: string } | undefined>()
+  const [setupIntent, setPaymentIntent] = useState<SetupIntent | null>()
+  const [isOpen, setIsOpen] = useState(!!clientSecret)
+
+  // Upgrade mutation
+  const [upgradeMutation] = useUpgradeToProfessionalPlanMutation({
+    variables: { planId },
+    refetchQueries: [namedOperations.Query.Subscription],
+    onCompleted: ret => {
+      const intent
+        = ret.createPlatformSubscription?.latestInvoice?.paymentIntent
+
+      console.log('paymentIntent', intent)
+      setUpgradeSuccess(true)
+    },
+    onError: error => {
+      setError(error)
+      console.log('Mutation error', error.message)
+    },
+  })
+
+  // Confirm payment
+  useEffect(() => {
+    if (stripe && clientSecret) {
+      console.log('stripe confirmCardPayment')
+      try {
+        stripe
+          .retrieveSetupIntent(clientSecret)
+          .then(({ setupIntent, error }) => {
+            console.log('stripe confirmCardPayment returned, ',
+              setupIntent,
+              error)
+
+            setError(error)
+            setPaymentIntent(setupIntent)
```

Comment:
we need to set this as the default payment method before upgrading so the subscription will actually bill the card

Reply:
I can add this to the createSubscription api too if it'll make things simpler or we can do it in the frontend

Reply:
(but generally the setup intent form should always add that payment method as the new default)